src/Integration.UnitTests/Binding/BindingControllerTests.cs

```diff
@@ -156,32 +153,51 @@ public void BindingController_BindCommand_Status()
         }
 
         [TestMethod]
-        public void BindingController_BindCommand_Status_VsState()
```

Comment:
I've split this test into three separate tests:
* one to test the UIContext flags (cases 1 and 2),
* one to test the non-managed project kind (case 3)
* one to test the "no projects" scenario (case 4)